src/reanimated2/layoutReanimation/sharedTransitions/ProgressTransitionManager.ts

```diff
@@ -149,6 +160,13 @@ function createProgressTransitionRegister() {
         progressAnimation!(viewTag, snapshot, progress);
       }
     },
+    onAndroidFinishTransitioning: () => {
+      if (toRemove.size === 0) {
```

Comment:
Let's make if conditional here same is in `onTransitionEnd` (toRemove.size > 0 and reverse order of clauses)